integration-tests/tests/src/utils/open-realm.ts

```diff
@@ -38,43 +38,61 @@ export async function openRealm(
   partialConfig: LocalConfiguration | SyncedConfiguration = {},
   user: User,
 ): Promise<{ config: Configuration; realm: Realm }> {
-  const nonce = new BSON.ObjectId().toHexString();
-  const path = `temp-${nonce}.realm`;
-
   if (!partialConfig.sync) {
-    const config = { ...partialConfig, path } as LocalConfiguration;
+    const config = createLocalConfig(partialConfig as LocalConfiguration);
     const realm = await Realm.open(config);
     return { config, realm };
-  } else {
-    const config = {
-      ...partialConfig,
-      path,
-      sync: {
-        user: user,
-        ...(partialConfig.sync.flexible ? { flexible: true } : { partitionValue: nonce }),
-        _sessionStopPolicy: "immediately",
-        ...partialConfig.sync,
-      },
-    } as Configuration;
-    const realm = await Realm.open(config);
+  }
 
-    // Upload the schema, ensuring a valid connection. uploadAllLocalChanges
-    // will not resolve with flexible sync enabled until we have created an
-    // initial subscription set, so skip it if we have a flexible config.
-    // There is an issue on the cloud side which will fix this.
-    if (!config.sync?.flexible) {
-      if (!realm.syncSession) {
-        throw new Error("No syncSession found on realm");
-      }
+  const config = createSyncConfig(partialConfig, user);
+  const realm = await Realm.open(config);
 
-      await realm.syncSession.uploadAllLocalChanges();
+  // Upload the schema, ensuring a valid connection. uploadAllLocalChanges
+  // will not resolve with flexible sync enabled until we have created an
+  // initial subscription set, so skip it if we have a flexible config.
+  // There is an issue on the cloud side which will fix this.
+  if (!config.sync?.flexible) {
+    if (!realm.syncSession) {
+      throw new Error("No syncSession found on realm");
     }
 
-    // TODO: This should probably be done in Realm.open()
-    if (config.sync?.flexible) {
-      await realm.subscriptions.waitForSynchronization();
-    }
+    await realm.syncSession.uploadAllLocalChanges();
+  }
 
-    return { config, realm };
+  // TODO: This should probably be done in Realm.open()
+  if (config.sync?.flexible) {
+    await realm.subscriptions.waitForSynchronization();
   }
+
+  return { config, realm };
+}
+
+export function createSyncConfig(partialConfig: SyncedConfiguration = {}, user: User): Configuration {
```

Comment:
If this function, as well as `createLocalConfig`, are only used in this file you can remove `export`. If they're needed outside of this file, you can add `/** @internal */` above their declarations 👍

Reply:
At the moment `createSyncConfig` is used in another file, while `createLocalConfig` isn't. I have extracted these two methods from the bigger method that was there before that was also creating a realm, as sometimes we just need a configuration. For this reason I have decided to export both anyways, as I expect they could be useful in some cases. 

Regarding `@internal`, this is the integration-test package, I suppose we don't need anything here, because this is not exposed to final users, am I wrong?

Reply:
I stand corrected! :) Annotation not needed in the integration tests 👍